### PR TITLE
fix(secret): database should use user-defined secret if specified

### PR DESCRIPTION
**Pull request by tthvo**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits): `git commit -S -m "YOUR_COMMIT_MESSAGE"`
_______________________________________________

Fixes: #826 

## Description of the change:

- Updated the controller to consider custom secret for the database container if specified.
- Tests: Added checks to validate storage, database and auth-proxy container specs. These are previously missing.
  - The AuthProxy container should have some checks for the container's arguments. For OpenShift Auth Proxy, most configs are specified using flags. Maybe, its better to split that into a separate follow-up PR. 

## Motivation for the change:

See #826. Tests are not yet validating the storage, database and auth-proxy containers.

## How to manually test:

Create a database secret:

```bash
cat <<EOF | oc apply -f -
apiVersion: v1
stringData:
  CONNECTION_KEY: connection_key
  ENCRYPTION_KEY: encryption_key
kind: Secret
metadata:
  name: db-secret
type: Opaque
EOF
```

Deploy the operator and create a cryostat CR with database secret specified:

```bash
export IMAGE_NAMESPACE=quay.io/some-user
export IMAGE_VERSION=test-version
make oci-build bundle bundle-build
podman push $IMAGE_NAMESPACE/cryostat-operator:$IMAGE_VERSION
podman push $IMAGE_NAMESPACE/cryostat-operator-bundle:$IMAGE_VERSION
make deploy_bundle

cat <<EOF | oc apply -f -
apiVersion: operator.cryostat.io/v1beta2
kind: Cryostat
metadata:
  name: cryostat-sample
spec:
  enableCertManager: true
  trustedCertSecrets: []
  eventTemplates: []
  storageOptions:
    pvc:
      labels: {}
      annotations: {}
      spec: {}
  reportOptions:
    replicas: 0
  databaseOptions:
    secretName: db-secret
EOF
```

The deployment should progress and all containers should be up and running.

### CODE REVIEW

**Review by andrewazores — COMMENTED** [review]
Looks great, thanks @tthvo !

File: internal/test/resources.go [hunk 1]
```diff
@@ -1654,6 +1827,49 @@ func (r *TestResources) NewDatasourceLivenessProbe() *corev1.Probe {
...
+	return &corev1.Probe{
+		ProbeHandler: corev1.ProbeHandler{
+			Exec: &corev1.ExecAction{
+				Command: []string{"pg_isready", "-U", "cryostat3", "-d", "cryostat3"},
+			},
+		},
+	}
+}
+
+func (r *TestResources) NewAuthProxyLivnessProbe() *corev1.Probe {
```
[the remark this answers is not in the document]
Comment (tthvo) [reply internal/test/resources.go:1853]: Opps :D

**Review by ebaron — APPROVED** [review]
Much appreciated! Thanks @tthvo!

**Pull request merged**